src/bigint.rs

```diff
@@ -874,6 +876,75 @@ impl BigUint {
 
     /// Creates and initializes a `BigUint`.
     ///
+    /// The bytes are in big-endian byte order
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use num::bigint::BigUint;
+    ///
+    /// assert_eq!(BigUint::from_bytes_be("A".as_bytes()),
+    ///            BigUint::parse_bytes("65".as_bytes(), 10).unwrap());
+    /// assert_eq!(BigUint::from_bytes_be("AA".as_bytes()),
+    ///            BigUint::parse_bytes("16705".as_bytes(), 10).unwrap());
+    /// assert_eq!(BigUint::from_bytes_be("AB".as_bytes()),
+    ///            BigUint::parse_bytes("16706".as_bytes(), 10).unwrap());
+    /// assert_eq!(BigUint::from_bytes_be("Hello world!".as_bytes()),
+    ///            BigUint::parse_bytes("22405534230753963835153736737".as_bytes(), 10).unwrap());
+    /// ```
+    #[inline]
+    pub fn from_bytes_be(bytes: &[u8]) -> BigUint {
+        if bytes.is_empty() {
+            Zero::zero()
+        } else {
+            BigUint::parse_bytes(bytes.to_hex().as_bytes(), 16).unwrap()
+        }
+    }
+
+    /// Creates and initializes a `BigUint`.
+    ///
+    /// The bytes are in little-endian byte order
+    #[inline]
+    pub fn from_bytes_le(bytes: &[u8]) -> BigUint {
+        let mut v = bytes.to_vec();
+        v.reverse();
+        BigUint::from_bytes_be(&*v)
+    }
+
+    /// Returns the byte representation of the `BigUint` in little-endian byte order.
+    #[inline]
+    pub fn to_bytes_le(&self) -> Vec<u8> {
```

Comment:
Possible extension: make this return an `Iterator` rather than returning a `Vec`.